source/MaterialXGenShader/Util.cpp

```diff
@@ -607,6 +855,14 @@ void findRenderableElements(ConstDocumentPtr doc, vector<TypedElementPtr>& eleme
             }
         }
     }
+}
+
+void findRenderableElements(ConstDocumentPtr doc, vector<TypedElementPtr>& elements, bool includeReferencedGraphs)
+{
+    std::unordered_set<OutputPtr> processedOutputs;
+
+    findRenderableMaterialNodes(doc, elements, includeReferencedGraphs, processedOutputs);
+    findRenderableShaderRefs(doc, elements, includeReferencedGraphs, processedOutputs);
 
     // Find node graph outputs. Skip any light shaders
     for (NodeGraphPtr nodeGraph : doc->getNodeGraphs())
```

Comment:
Below if nodeGraph->hasSourceUri() is false, do we need to check output->hasSourceUri() further below, since it's an output on the same nodeGraph?

Reply:
It's theoretically possible but doesn't make sense in practice. I'll remove the output check.